models/ldndc/R/met2model.LDNDC.R

```diff
@@ -0,0 +1,247 @@
+      
+    
+      # Calculate the time steps
+      # 3600 * 24 = 86 400 (seconds in a day)
+      tstep <- 86400/(sec[2]- sec[1])
+      
+      
+      
+      
+      ## Determine the simulation days and calculate the start and end indexes
+      ## by using start_index and end_index functions. These functions are found
+      ## at the end of this file
+      
+      if(year != start_year & year != end_year){  
+        
+        # Simulation days
+        simu_days <- 1:PEcAn.utils::days_in_year(year)
+        
+        # Whole year, from first to last index
+        ind_s <- 1
+        ind_e <- which.max(sec)
+        
+      }else if(year == start_year & year != end_year){
+        
+        # Simulation days
+        simu_days <- seq(lubridate::yday(start_date),PEcAn.utils::days_in_year(year))
+        
+        # Check the function to see how the start index is calculated
+        ind_s <- start_index(units, start_date, sec)
+        ind_e <- which.max(sec)
+        
+      }else if(year != start_year & year == end_year){
+        
+        # Simulation days
+        simu_days <- seq(1:lubridate::yday(end_date))
+        
+        # Check the function to see how the end index is calculated
+        ind_s <- 1
+        ind_e <- end_index(units, start_date, end_date, sec, tstep)
+        
+      }else{
+        
+        # Simulation days
+        simu_days <- seq(lubridate::yday(start_date), lubridate::yday(end_date))
+        
+        # Need to calculate both first and last index by using functions
+        ind_s <- start_index(units, start_date, sec)
+        ind_e <- end_index(units, start_date, end_date, sec, tstep)
+        
+      }
+      
+      # Based on the simulation days, this is year, day and subday info
+      # for model to use.
+      y <- rep(year, length(simu_days)*tstep) #year
+      d <- rep(simu_days, each = tstep) #day
+      subd <- rep(1:tstep, length(simu_days)) #subday
+      
+      # On above the starting and ending indexes has been determined.
+      # Here those are just collapsed.
+      ind <- ind_s:ind_e
+      
+        
+      
+      ## Info for climate file that model is using ##
+      # Latitude and longitude
+      lat <- ncdf4::ncvar_get(nc, "latitude")
+      lon <- ncdf4::ncvar_get(nc, "longitude")
+      
+      
+      # Average air temperature
+      Tair <-ncdf4::ncvar_get(nc, "air_temperature")[ind]  ## in Kelvin
+      tavg <- udunits2::ud.convert(Tair, "K", "degC")
+      
+      # Wind speed
+      wind <- try(ncdf4::ncvar_get(nc, "wind_speed"))[ind]
+      
+      # Precipation
+      prec <- ncdf4::ncvar_get(nc, "precipitation_flux")[ind]
+      
+      # Global radiation
+      grad <- ncdf4::ncvar_get(nc, "surface_downwelling_shortwave_flux_in_air")[ind]  ## in W/m2
+      
+      # Vapor Pressure Deficit Pa ---> kPa, need to ne divided by 1000
+      VPD <- try(ncdf4::ncvar_get(nc, "water_vapor_saturation_deficit"))[ind]  ## in Pa
```

Comment:
If these don't exist (i.e. no such variable in the nc) would division by 1000 or any number break this code (below)? Would it break the model simulation if there is no VPD in the model climate file? Maybe add checks accordingly, if VPD is absent and important for the model, either fail here already or consider some derivation, e.g. see [here](https://github.com/PecanProject/pecan/blob/develop/models/sipnet/R/met2model.SIPNET.R#L174)